assets/src/block-editor/AssignOnlyFlatTermSelector/index.js

```diff
@@ -0,0 +1,187 @@
+
+  const {
+    terms,
+    taxonomy,
+    hasAssignAction,
+    hasResolvedTerms,
+  } = useSelect(
+    select => {
+      const {getCurrentPost, getEditedPostAttribute} = select('core/editor');
+      const {getEntityRecords, getTaxonomy, hasFinishedResolution} = select(coreStore);
+      const post = getCurrentPost();
+      const _taxonomy = getTaxonomy(slug);
+      const _termIds = _taxonomy ? getEditedPostAttribute(_taxonomy.rest_base) : EMPTY_ARRAY;
+
+      const query = {
+        ...DEFAULT_QUERY,
+        include: _termIds.join(','),
+        per_page: -1,
+      };
+
+      return {
+        hasAssignAction: _taxonomy ? post._links?.['wp:action-assign-' + _taxonomy.rest_base] ?? false : false,
+        taxonomy: _taxonomy,
+        termIds: _termIds,
+        terms: _termIds.length ? getEntityRecords('taxonomy', slug, query) : EMPTY_ARRAY,
+        hasResolvedTerms: hasFinishedResolution('getEntityRecords', ['taxonomy', slug, query]),
+      };
+    },
+    [slug]
+  );
+
+  const {searchResults} = useSelect(
+    select => {
+      const {getEntityRecords} = select(coreStore);
+
+      return {
+        searchResults: !!search ? getEntityRecords('taxonomy', slug, {
+          ...DEFAULT_QUERY,
+          search,
+        }) : EMPTY_ARRAY,
+      };
+    },
+    [search, slug]
+  );
+
+  // Update terms state only after the selectors are resolved.
+  // We're using this to avoid terms temporarily disappearing on slow networks
+  // while core data makes REST API requests.
+  useEffect(() => {
+    if (hasResolvedTerms) {
+      const newValues = (terms ?? []).map(term =>
+        decodeEntities(term.name)
+      );
+
+      setValues(newValues);
+    }
+  }, [terms, hasResolvedTerms]);
+
+  const suggestions = useMemo(() => {
+    return (searchResults ?? []).map(term =>
+      decodeEntities(term.name)
+    );
+  }, [searchResults]);
+
+  const {editPost} = useDispatch('core/editor');
+
+  if (!hasAssignAction) {
+    return null;
+  }
+
+  function onUpdateTerms(newTermIds) {
+    editPost({[taxonomy.rest_base]: newTermIds});
+  }
+
+  function onChange(termNames) {
+    const availableTerms = [
+      ...(terms ?? []),
+      ...(searchResults ?? []),
+    ];
+
+    const uniqueTerms = termNames.reduce((acc, name) => {
+      if (
+        !acc.some(n => n.toLowerCase() === name.toLowerCase())
+      ) {
+        acc.push(name);
+      }
+      return acc;
+    }, []);
+
+    // Filter to remove new terms since we don't allow creation.
+    const allowedTerms = uniqueTerms.filter(
+      termName => availableTerms.find(term => isSameTermName(term.name, termName))
+    );
+
+    setValues(allowedTerms);
+
+    return onUpdateTerms(termNamesToIds(allowedTerms, availableTerms));
+  }
+
+  const newTermLabel =
+    taxonomy?.labels?.add_new_item ??
+    (slug === 'post_tag' ? __('Add new tag') : __('Add new Term'));
+  const singularName =
+    taxonomy?.labels?.singular_name ??
+    (slug === 'post_tag' ? __('Tag') : __('Term'));
```

Comment:
Don't we need to add the `domain`?

Reply:
No, because this is copied from a WordPress original file. We should update it though, and there is a [ticket](https://jira.greenpeace.org/browse/PLANET-7728) for that already!